### ISSUE TITLE
Form could not be imported. File type must be JSON.

### ISSUE THREAD

**Issue by alvarolagarde** [CalderaWP/Caldera-Forms#3137]
# PLEASE READ THIS LINE & Then Delete - We need this information to best understand your issue, please answer all questions!

## Do You Need Immediate Help?
* If so, please open a support ticket at https://CalderaForms.com/support

## Is This A Feature Request?
* If so, that's fine, please describe your desired featured and delete the rest of this template.

## What Version Of Caldera Forms, WordPress and PHP Are You Using?
*You can copy the "Short Debug" line from the Caldera Forms support menu's debug tab

## What Is The Unexpected Behaviour?
* Please explain what you did, what you expected to happen, and what happened instead.

## What PHP Errors Have You Logged While Reproducing This Bug?
* See: https://calderaforms.com/2016/05/wordpress-debug-logging/

## What JavaScript Errors Have You Seen While Reproducing This Bug?
* See: http://webmasters.stackexchange.com/questions/8525/how-to-open-the-javascript-console-in-different-browsers

**Comment by iTrooper5118** [comment]
Yep, I've pretty much gotten the same error two and there doesn't seem to be any useful info on how to fix this which is preventing me from rebuilding/porting the forms over from another site.

**Comment by lynxlynxlynx** [comment]
Same here, just trying to move the form from a multisite subsite, so fresh export, same plugin version, but import fails with the same reason.

$type_check[ 'ext'] is empty. 5.1 added a $real_mime check to the wp_check_filetype_and_ext  function, so perhaps that's tripping it up.

**Comment by lynxlynxlynx** [comment]
Yeah, the new change broke it, it's hitting the branch with this comment:

> Everything else including image/* and application/*:
> If the real content type doesn't match the file extension, assume it's dangerous.

And it happens since $type is text/plain, but $real_mime is text/html (probably due to some of the fields containing HTML). A similar [regression with csv](https://core.trac.wordpress.org/ticket/47448) that also links to a wider core discussion.

I'm not sure this can be fixed completely by the plugin, since if you add 'text/html' to the mime list in `classes/admin.php:save_form()` it just trips up the wp function later if you don't have "unfiltered_html" capabilities. It checks the mime against `get_allowed_mime_types()` which excludes html if you lack that [capability](https://wordpress.org/support/article/roles-and-capabilities/#unfiltered_html).

So easily fixable for admin users, not for anyone else that may have access to importing forms.

**Comment by Kwagga-Quagga** [comment]
Hi @lynxlynxlynx 

I understood that you foud a way through ? 

> So easily fixable for admin users, not for anyone else that may have access to importing forms.
>
> TLDR: if you want a quick hack to get the import done, just temporarily comment out the wp_die call with the title error message.

Could you please provide some more detailed explanation ?
I can live with any of the two solution but I can't figure out what do I have exactly to do with any you mention.

Also there is [this post that suggest a solution](https://translate.google.de/translate?sl=de&tl=en&u=https%3A%2F%2Ftorstenlandsiedel.de%2F2019%2F01%2F28%2Fkaputten-import-bei-caldera-forms-reparieren%2F) but it do not work either for me.

Thank's  in advance

**Comment by Kwagga-Quagga** [comment]
> Just search for the error message in the plugin code and disable that little problematic block by commenting it out.

Excellent, It works indeed

Thank's

If one need more help, [it's here](https://github.com/CalderaWP/Caldera-Forms/search?q=Form+could+not+be+imported.+File+type+must+be+JSON.&unscoped_q=Form+could+not+be+imported.+File+type+must+be+JSON.)

**Comment by mwittmann** [comment]
Tracing through the code, it turns out that a plain text `*.json` file will be accepted as long as WordPress allows you to upload files with MIME type "`text/plain`". By default, WordPress does not! There are several ways to remedy this. For me, running WordPress multisite, all I had to do was add the "`txt`" extension in the multisite configuration page "Network Admin... Settings" Look for the "Upload Settings" section and add `txt` to the list of allowed "Upload file types".

**Comment by brokermodel** [comment]
> Tracing through the code, it turns out that a plain text `*.json` file **will** be accepted as long as WordPress allows you to upload files with MIME type "`text/plain`". By default, WordPress does not! There are several ways to remedy this. For me, running WordPress multisite, all I had to do was add the "`txt`" extension in the multisite configuration page "Network Admin... Settings" Look for the "Upload Settings" section and add `txt` to the list of allowed "Upload file types".

Wow! this was an easy fix and very intuitive fix - worked for me. Thank you.

**Comment by grrog** [comment]
> Just search for the error message in the plugin code and disable that little problematic block by commenting it out.

Thank you very much for that tip! ;)

**Comment by mbaierl** [comment]
A year old and not fixed 🙈

**Comment by aronmoshe-m** [comment]
Any change of getting this fixed instead of needing a hack-y temporary solution?

**Comment by aronmoshe-m** [comment]
Still would love to see this fixed in an actual release, as a paying customer.

**Comment by PaulBarbieux** [comment]
Same problem.
Function json_last_error_msg() shows: Syntax error.

**Comment by Zodiac1978** [comment]
Why not just add `text/html` as a third allowed mime type next to `application/json` and `text/plain`:

https://github.com/CalderaWP/Caldera-Forms/blob/06974585640525df669433709f004afc2e3735cf/classes/admin.php#L1303-L1310

The mime type detection on most *nix servers will be signature based and seems to be not very reliable for JSON files, which makes sense, if you look at a JSON file. It can be easily misinterpreted as plain text or HTML file (especially if you have added many HTML fields with text).

**Comment by marcocalcanti** [comment]
Just remove this code:

`if( ! 'json' === $type_check[ 'ext'] ){
						wp_die( esc_html__( 'Form could not be imported. File type must be JSON.', 'caldera-forms' ) );
					}`

**Comment by uwclive** [comment]
I know this is an old thread, but it still an issue in the community supported fork of Caldera Forms, so I'm posting in case it helps.

I tried to get around this issue by adding:

`define( 'ALLOW_UNFILTERED_UPLOADS', true );`

to the wp-config.php file, but it did not work.  However, what does work is installing the plugin "WP Add Mime Types".  I added this to the "added values" section of the plugin settings:

`json	= text/plain`

The json file imported without errors.  The plugin does avoid the attempt to determine the real file type of a file by WordPress core, so that could be a concern.  I don't really like installing a plugin for something that is so infrequently needed, but at least it can be deactivated or even uninstalled after use.
